File: 연습문제/LV.1/자연수 뒤집어 배열로 만들기.py
#LV.1 자연수 뒤집어 배열로 만들기
#주어진 자연수를 뒤집어서 리스트변환시키는 알고리즘
#
#string 변환후 [::-1]으로 뒤집기! 그후, 리스트에 append()시키기 :) 
#

def solution(n):
    answer = []
    
    tmp = str(n)
    tmp = tmp[::-1]
    
    for i in tmp : 
            answer.append(int(i))
    
    return answer

File: 연습문제/LV.1/자연수 뒤집어 배열로 만들기.java
import java.util.*;


class Solution {
    public int[] solution(long n) {
        
        String s = String.valueOf(n);
        StringBuilder sb = new StringBuilder(s);
        // stringbuilder은 불변의 객체인 string과 다르게 동적으로 할당가능하다
        
        
        sb = sb.reverse();
        System.out.println(sb); 
        String[] ss = sb.toString().split("");
        System.out.println(Arrays.toString(ss)); 
        
        int[] answer = new int[ss.length];
        for (int i=0; i<ss.length; i++){
            answer[i] = Integer.parseInt(ss[i]);
            // 숫자형의 문자열을, 첫번째 인자값으로 받고, 변환할 진수값을 입력하면
            // 해당 진수에 맞춰 Integer 형 반환
        }
        
        
        return answer ; 
    }
}
